src/NUnitFramework/framework/Constraints/EmptyConstraint.cs

```diff
@@ -48,11 +50,13 @@ public override string Description
         /// <returns>True for success, false for failure</returns>
         public override ConstraintResult ApplyTo<TActual>(TActual actual)
         {
-            // NOTE: actual is string will fail for a null typed as string
-            if (typeof(TActual) == typeof(string))
+            // NOTE: actual is string will fail for a null typed as string           
+            Type actualType = typeof(TActual);
+            
+            if (actualType == typeof(string))
                 realConstraint = new EmptyStringConstraint();
             else if (actual == null)
-                throw new System.ArgumentException("The actual value must be a string or a non-null IEnumerable or DirectoryInfo", nameof(actual));
+                throw new System.ArgumentException($"The actual value of type - {actualType} must be a string or a non-null IEnumerable or DirectoryInfo", nameof(actual));
```

Comment:
The changes all look good! I think this error message is getting a bit long-winded however - could I suggest rewording it to something a bit more natural? How about something like...

```
$"The actual value must be a string, non-null IEnumerable or DirectoryInfo. The value passed was of type {actualType}."
```

Reply:
Yes. For reference, [this is message](https://sharplab.io/#v2:EYLgHgbALANALiATgVwHYB9kGcCWqDmABAMoCeWcApgLYDcAsAFBMAOywANjgMaEUCGcHoW4d+WLIQAKiAPb5E/akwDeTQhsJtOwgUN4A3WTgAmhALL88ACgCU6zWsaaXhAMKzUWWR0oA6AHVEHCoAGTxKa1RKAHdCAEFEfGRqSlQ4AHlkTIAzACV+AkoAUTBuShYhT2sAIhZ+RWpCVCVKGphCKAAmDprUiX58Nts/c0oBodsGZ00AXyZZoA) the `ArgumentOutOfRangeException` `actualValue` parameter causes:

> Actual value was 42.